src/main/groovy/com/ullink/Msbuild.groovy

```diff
@@ -122,11 +123,15 @@ class Msbuild extends ConventionTask {
     }
 
     boolean resolveProject() {
-        if (projectParsed == null) {
+        if (projectParsed == null && parseProject == true) {
             if (isSolutionBuild()) {
                 def result = parseProjectFile(getSolutionFile())
                 allProjects = result.collectEntries { [it.key, new ProjectFileParser(msbuild: this, eval: it.value)] }
                 projectParsed = allProjects[getProjectName()]
+                if (projectParsed == null) {
+                    parseProject = false
```

Comment:
above method getProjects will return null in that case. As projects have been parsed and are available I think it'll be better to return the actual list, no ?